Content.Shared/_FarHorizons/Power/Generation/FissionGenerator/SharedReactorPartSystem.cs

```diff
@@ -0,0 +1,339 @@
+            return;
+
+        using (args.PushGroup(nameof(ReactorPartComponent)))
+        {
+            switch(comp.NRadioactive)
+            {
+                case > 8:
+                    args.PushMarkup(Loc.GetString("reactor-part-nrad-5"));
+                    break;
+                case > 6:
+                    args.PushMarkup(Loc.GetString("reactor-part-nrad-4"));
+                    break;
+                case > 4:
+                    args.PushMarkup(Loc.GetString("reactor-part-nrad-3"));
+                    break;
+                case > 2:
+                    args.PushMarkup(Loc.GetString("reactor-part-nrad-2"));
+                    break;
+                case > 1:
+                    args.PushMarkup(Loc.GetString("reactor-part-nrad-1"));
+                    break;
+                case > 0:
+                    args.PushMarkup(Loc.GetString("reactor-part-nrad-0"));
+                    break;
+            }
+
+            switch (comp.Radioactive)
+            {
+                case > 8:
+                    args.PushMarkup(Loc.GetString("reactor-part-rad-5"));
+                    break;
+                case > 6:
+                    args.PushMarkup(Loc.GetString("reactor-part-rad-4"));
+                    break;
+                case > 4:
+                    args.PushMarkup(Loc.GetString("reactor-part-rad-3"));
+                    break;
+                case > 2:
+                    args.PushMarkup(Loc.GetString("reactor-part-rad-2"));
+                    break;
+                case > 1:
+                    args.PushMarkup(Loc.GetString("reactor-part-rad-1"));
+                    break;
+                case > 0:
+                    args.PushMarkup(Loc.GetString("reactor-part-rad-0"));
+                    break;
+            }
+
+            if (comp.Temperature > Atmospherics.T0C + 500)
+                args.PushMarkup(Loc.GetString("reactor-part-burning"));
+            else if (comp.Temperature > Atmospherics.T0C + 80)
+                args.PushMarkup(Loc.GetString("reactor-part-hot"));
+        }
+    }
+
+    public override void Update(float frameTime)
+    {
+        _accumulator += frameTime;
+        if (_accumulator > _threshold)
+        {
+            AccUpdate();
+            _accumulator = 0;
+        }
+    }
+
+    private void AccUpdate()
+    {
+        var query = EntityQueryEnumerator<ReactorPartComponent>();
+        while (query.MoveNext(out var uid, out var component))
+        {
+            if (!_entityManager.HasComponent<RadiationSourceComponent>(uid))
+            {
+                var radcomp = EnsureComp<RadiationSourceComponent>(uid);
+                radcomp.Intensity = (component.Radioactive * 0.1f) + (component.NRadioactive * 0.15f) + (component.SpentFuel * 0.125f);
+            }
+
+            if (component.NRadioactive > 0 && !_entityManager.HasComponent<PointLightComponent>(uid))
+            {
+                var lightcomp = _lightSystem.EnsureLight(uid);
```

Comment:
So here's the single use of server component that you're including entire engine for. It can easily be replaced with

```suggestion
            if (component.NRadioactive > 0 && !_lightSystem.TryGetLight(uid, out _))
            {
                var lightcomp = _lightSystem.EnsureLight(uid);
```